src/validators/validators.ts

```diff
@@ -10,4 +10,9 @@ export const isBoolean = validator(
   "must be a boolean"
 );
 
+export const isUuid = validator(
+  val => val.replace(/-/g, "").match(/^[0-9a-f]{32}/i),
```

Comment:
Since this one contains the start notation, should we also include end `$`?

I am uncertain about the removal of dashes. What we see / test above is the standard string representation, which is 8-4-4-4-12. I would rather fix it at this standard string representation, or alternatively, no dashes.

```ts
val => val.match(/^[0-9a-f]{8}\-?[0-9a-f]{4}\-?[0-9a-f]{4}\-?[0-9a-f]{4}\-?[0-9a-f]{12}$/i),
```

But this is just my thinking - may be way off here and you already have specific use cases in mind.

Reply:
Oop, good catch with the `$`!

You're totally right about the canonical UUID representation. But the most important concept provider, [CIEL](https://openconceptlab.org/orgs/CIEL/sources/CIEL/concepts/123118/) uses 32-digit alphanumeric strings as "External IDs" and these are very widely used in OpenMRS as if they're UUIDs.

Ooh, but actually, maybe it would be better to check if the UUID has the dashed form or the undashed form, and not something strange like 8-12-12.

Reply:
Oh no, CIEL actually uses 36 digit alphanumeric strings! Which means, actually, I don't understand why these tests were passing...